File: lcnaf.ts
export const LCNAFPattern: string = "^[a-zA-Z0-9]+$"; // Example pattern for LCNAF IDs
export const reLCNAF: RegExp = new RegExp(LCNAFPattern);

/**
 * normalizeLCNAF
 * @param id, a string holding the LCNAF identifier
 * @returns normalized LCNAF identifier as a string.
 */
export function normalizeLCNAF(id: string): string {
  return id.trim();
}

/**
 * validateLCNAF
 * @param id, a string holding the LCNAF identifier
 * @returns true if valid, false otherwise
 */
export function validateLCNAF(id: string): boolean {
  const normalizedId = normalizeLCNAF(id);
  return reLCNAF.test(normalizedId);
}

File: lcnaf.go
package metadatatools

import (
	"regexp"
	"strings"
)

// LCNAFPattern matches a normalized Library of Congress Name Authority
// File identifier.
const LCNAFPattern = `^[a-zA-Z0-9]+$`

// reLCNAF is the compiled form of LCNAFPattern.
var reLCNAF = regexp.MustCompile(LCNAFPattern)

// NormalizeLCNAF trims whitespace from id.
func NormalizeLCNAF(id string) string {
	return strings.TrimSpace(id)
}

// ValidateLCNAF normalizes id then validates it against LCNAFPattern.
func ValidateLCNAF(id string) bool {
	return reLCNAF.MatchString(NormalizeLCNAF(id))
}
